cscs-checks/system/basic/file_system_check.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class FileSystemCommandCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        self.descr = 'File system sanity test base'
+        # TODO: test from cn as well
+        self.valid_systems = ['daint:login', 'dom:login']
+        self.valid_prog_environs = ['builtin']
+        self.num_tasks = 1
+        self.num_tasks_per_node = 1
+        self.sanity_patterns = sn.assert_found(r'0', self.stdout)
```

Comment:
I would rather use the exit code from the job instead of using the `postrun_cmds`:

```suggestion
        self.sanity_patterns = sn.assert_eq(sn.getattr(self, 'job').exitcode, 0)
```

Reply:
I'm getting the error 
AttributeError: '_DeferredExpression' object has no attribute 'exitcode' (or other job fields like jobid)
with the above sanity_pattern definition in the init function or in a method decorated with @rfm.run_before('sanity')

Reply:
Oh yes, my bad. You should write it as follows:

```suggestion
@rfm.run_before('sanity')
def set_sanity(self):
        self.sanity_patterns = sn.assert_eq(self.job.exitcode, 0)
```

Reply:
Done